### PR TITLE
implements MessageEndpoint structure

### LINKED ISSUE #57: [message_endpoint] sketch the structure
`message_endpoint` name is referenced from: https://english.stackexchange.com/questions/369828/what-is-the-generic-term-for-sender-and-receiver
select **endpoint** to represent both sender and receiver.

`MessageEndpoint` basically do receiving packet and determine which logic executed based on the packet. Additionally `MessageEndpoint` call `MemberDelgator` to update member status, and encrypt data

### PR DESCRIPTION
**Pull request by zeroFruit**
resolved: #57 

details:

define MessageEndpoint structure

`MessageEndpoint` basically do receiving packet and determine which logic executed based on the packet. Additionally `MessageEndpoint` call `MemberDelgator` to update member status, and encrypt data

### CODE REVIEW

File: message_endpoint.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package swim
+
+// I think this Message struct should come from pb?
+type Message struct{}
+
```
Comment (junbeomlee) [inline message_endpoint.go:11]: Yes, use `protoc` to create message.go

File: message_endpoint.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// MessageEndpoint basically do receiving packet and determine
+// which logic executed based on the packet.
+type MessageEndpoint struct {
+	transport UDPTransport
+}
+
+// ListenPacket is a log running goroutine that pulls packet from the
+// transport and pass it for processing
+func (m *MessageEndpoint) ListenPacket() {
+
```
Comment (junbeomlee) [inline message_endpoint.go:56]: I think just `Listen` or `Start` is fine, I want to hide `Packet` because it is a low abstraction level.

Comment (zeroFruit) [reply message_endpoint.go:56]: `Listen` would be good!

File: message_endpoint.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+// processPacket process given packet, this procedure may include
+// decrypting data and converting it to message
```
Comment (junbeomlee) [inline message_endpoint.go:76]: Start with capital letters.
`ProcessPacket`

File: message_endpoint.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package swim
+
+// I think this Message struct should come from pb?
+type Message struct{}
+
+// MessageEndpoint basically do receiving packet and determine
+// which logic executed based on the packet.
+type MessageEndpoint struct {
+	transport UDPTransport
+}
```
Comment (junbeomlee) [inline message_endpoint.go:47]: This struct should have delegate interface.

**Comment by junbeomlee** [comment]
And we need a way to send and receive packets synchronously!!
MemberList has it!

File: message_endpoint.go (hunk 3 above)
Comment (zeroFruit) [reply message_endpoint.go:76]: I think there's no need to expose this function

Comment (junbeomlee) [reply message_endpoint.go:76]: I mean the comment!!
`// processPacket process given packet, this procedure may include`

**Comment by zeroFruit** [comment]
clean up the commit logs

**Comment by zeroFruit** [comment]
I know that there's some missing part in:
```go
go func() {
    _, err := m.transport.WriteTo(d, addr)
    ...
}()
```
:-(

**Pull request closed without merging**